src/scene/materials/standard-material.js

```diff
@@ -768,19 +788,25 @@ function _defineTex2D(name, uv, channels, defChannel, vertexColor, detailMode) {
     // construct the transform uniform
     const mapTiling = `${name}MapTiling`;
     const mapOffset = `${name}MapOffset`;
+    const mapRotation = `${name}MapRotation`;
     const mapTransform = `${name}MapTransform`;
     defineUniform(mapTransform, (material, device, scene) => {
         const tiling = material[mapTiling];
         const offset = material[mapOffset];
-        if (tiling.x === 1 && tiling.y === 1 && offset.x === 0 && offset.y === 0) {
+        const rotation = material[mapRotation];
+
+        if (tiling.x === 1 && tiling.y === 1 &&
+            offset.x === 0 && offset.y === 0 &&
+            rotation === 0) {
             return null;
         }
 
-        const uniform = material._allocUniform(mapTransform, () => new Float32Array(4));
-        uniform[0] = tiling.x;
-        uniform[1] = tiling.y;
-        uniform[2] = offset.x;
-        uniform[3] = 1.0 - tiling.y - offset.y;
+        const uniform = material._allocUniform(mapTransform, () => new Float32Array(9));
+        uniform.set([
```

Comment:
don't create temp array here

Reply:
also, evaluate sin and cos only one time instead of twice

Reply:
The other option would be to store 5 uniforms. 4 as currently for offset and scale, and a single for rotation. And generate sin/cos and all that on the GPU. But that's probably not worth the effort.

Reply:
I'm not sure why `set` for matrix classes was written to take an array as a parameter. It should just take the individual elements IMHO, just like for vectors. And a `setFromArray` function could be added for the case where somebody wants to pass data that way.

Reply:
I think this set is Float32Array.set, which takes array. We should set individual entries of the array without the extra temp array.